test/spec_helper.rb

```diff
@@ -87,6 +87,17 @@ def require(path)
       allow(Yast::Arch).to receive(:ppc).and_return(architecture == :ppc)
       allow(Yast::Arch).to receive(:s390).and_return(architecture == :s390)
       allow(Yast::Arch).to receive(:aarch64).and_return(architecture == :aarch64)
+
+      # If :storage_arch is defined, the Storage::Arch object is used instead of Yast::Arch.
+      if respond_to?(:storage_arch)
+        arch = Y2Storage::Arch.new(storage_arch)
+
+        allow(Y2Storage::Arch).to receive(:new).and_return(arch)
+
+        allow(storage_arch).to receive(:x86?).and_return(architecture == :x86)
+        allow(storage_arch).to receive(:ppc?).and_return(architecture == :ppc)
+        allow(storage_arch).to receive(:s390?).and_return(architecture == :s390)
```

Comment:
why is not there all storage forwarded methods?

Reply:
These are the "architecture" methods. The other two (:efiboot? and :ppc_power_nv?) do not define the architecture itself, but properties related to the architecture. For those two properties, each tests mock them if it is needed, for example:

```
let(:architecture) { :x86 }

before do 
  allow(:storage_arch).to receive(:efiboot?).and_return(true)
end
```

Reply:
so it does not make sense to have some default like false here? to avoid call to storage and potential arch dependent test?